rtrlib/pfx/trie/trie-pfx.c

```diff
@@ -133,18 +153,31 @@ inline struct trie_node *pfx_table_get_root(const struct pfx_table *pfx_table, c
 
 int pfx_table_del_elem(struct node_data *data, const unsigned int index)
 {
+    struct data_elem *tmp;
+    struct data_elem deleted_elem = data->ary[index];
```

Comment:
well I know that C doesn't protect you from doing something stupid.

Nevertheless, the assert might come in handy for debugging. IMO an assert also makes it clear which values are expected for parameters. Hence, `assert(data != NULL)` could be added here, too.

However, we might discuss this separately - IMO asserts are useful and we should think about adding more often throughout the code. In a release build they are stripped anyway, so no code bloat ...